stack/services/src/main/java/org/apache/usergrid/security/tokens/cassandra/TokenServiceImpl.java

```diff
@@ -789,12 +789,17 @@ MetricsFactory getMetricsFactory() {
     }
 
 
-    private boolean isExternalSSOProviderEnabled() {
+    public boolean isExternalSSOProviderEnabled() {
         return Boolean.valueOf(properties.getProperty( USERGRID_EXTERNAL_SSO_ENABLED ));
     }
 
     private String getExternalSSOProvider(){
-        return properties.getProperty(USERGRID_EXTERNAL_PROVIDER);
+        try {
+            return properties.getProperty(USERGRID_EXTERNAL_PROVIDER);
+        }
+        catch(NullPointerException e ){
```

Comment:
I don't think this will throw an NPE as much as it would return ```null``` to the caller of ```getExternalSSOProvider()```